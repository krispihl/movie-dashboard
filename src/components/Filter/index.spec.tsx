import { Filter } from "./index";
import { render, screen, act } from '@testing-library/react';
import userEvent from "@testing-library/user-event";

jest.mock('../../contexts/movies', () => ({
    useMovies: () => ({
        genres: [
            { id: 1, name: 'All movies', popular: true },
            { id: 35, name: 'Comedy', popular: true },
            { id: 18, name: 'Drama', popular: true },
            { id: 80, name: 'Crime', popular: false },
            { id: 36, name: 'History', popular: false },
        ],
        setGenres: jest.fn(),
        activeGenre: 1,
        setActiveGenre: jest.fn(),
    })
}));

describe('src/components/Filter', () => {
    it('should render Filter correctly with popular genres', async () => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => {return { genres: [] }}
            })
        ) as jest.Mock;

        const { container } = render(<Filter />);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('More')).toBeInTheDocument();
        expect(screen.queryByText('History')).not.toBeInTheDocument();
    });

    it('should render Filter correctly with all genres', async () => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => {return { genres: [] }}
            })
        ) as jest.Mock;

        let container;

        await act(async () => {
            container = render(<Filter />).container;
        });

        const moreButton = screen.getByText('More');

        await act(async () => {
            userEvent.click(moreButton);
        });

        expect(container).toMatchSnapshot();
        expect(screen.queryByText('More')).not.toBeInTheDocument();
        expect(screen.getByText('Less')).toBeInTheDocument();
        expect(screen.getByText('History')).toBeInTheDocument();

        const allFilterButtons = document.querySelectorAll('.filter__item');

        expect(allFilterButtons).toHaveLength(6);
    });
});
