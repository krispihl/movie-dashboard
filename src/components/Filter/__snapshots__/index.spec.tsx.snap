// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Filter should render Filter correctly with all genres 1`] = `
<div>
  <ul
    class="genres"
  >
    <button
      class="genres__item genres__item--active"
    >
      All movies
    </button>
    <button
      class="genres__item"
    >
      Comedy
    </button>
    <button
      class="genres__item"
    >
      Drama
    </button>
    <button
      class="genres__item"
    >
      Crime
    </button>
    <button
      class="genres__item"
    >
      History
    </button>
    <button
      class="genres__item"
    >
      Less
    </button>
  </ul>
</div>
`;

exports[`src/components/Filter should render Filter correctly with popular genres 1`] = `
<div>
  <ul
    class="genres"
  >
    <button
      class="genres__item genres__item--active"
    >
      All movies
    </button>
    <button
      class="genres__item"
    >
      Comedy
    </button>
    <button
      class="genres__item"
    >
      Drama
    </button>
    <button
      class="genres__item"
    >
      More
    </button>
  </ul>
</div>
`;
